### ISSUE TITLE
HTMLHint (or extension support)

### ISSUE THREAD

**Issue by ptyork** [microsoft/monaco-editor#34]
Currently VSCode/Monaco supports linting for a good number of languages, but HTML is not one of them. VSCode makes up for this with an extension: https://github.com/Microsoft/vscode-htmlhint

So, will VSCode extensions (perhaps with minor mods) work in Monaco? If not, any plans to add HTML linting?

**Comment by alexandrudima** [comment]
fyi @aeschli 

Today the html language support is still inside `monaco-editor-core` aka in the vscode repo. That is because 100% the same code ships with vscode and with the editor. We plan to break up this code, to create a vscode html extension and once that is done, we can create a `monaco-html` editor plugin that reuses some of the code, but that is independent of vscode. We can then consider adding more features to it.

TL;DR we can't do it yet due to it affecting vscode, but we plan to fix this by making things separate.

**Comment by ptyork** [comment]
Sounds good, thanks. I'm creating a tool to help students learn HTML+CSS so this will be quite helpful. Happy to help with this when it becomes possible.

**Comment by joeskeen** [comment]
@alexandrudima would the creation of a `monaco-html` editor plugin include support for Emmet?

**Comment by scniro** [comment]
@alexandrudima has there been any positive progress on this? I as well and trying to take advantage of syntax support while editing HTML.

**Comment by scniro** [comment]
May I add that I see the [monaco-html](https://github.com/Microsoft/monaco-html) plugin. @alexandrudima is there I way I can accomplish the hinting with this module? I'm browsing the object and `monaco.languages.html.htmlDefaults` appears to be missing `diagnosticsOptions` similar to the other languages. Am I overlooking something?

**Comment by aeschli** [comment]
The Monaco HTML support doesn't include any validation and there are currently no plans to add it.

**Comment by scniro** [comment]
@aeschli may I ask, why? Is validating HTML just outright not interesting to Microsoft? I open up VS Code and have no sense of assurance that any of my markup is even close to being valid. I find this very surprising.

**Comment by aeschli** [comment]
It just hasn't gotten any priority yet and surprisingly there haven't been many requests for it. My guess this has to do that there are plenty of good validators out there.
It's quite some work to keep a validator up-to-date. One issue with HTML is that there are many templating variants of HTML around, and for these files validation often flags problems that are false positives.

**Comment by scniro** [comment]
@aeschli gotcha. I was hoping there could be at least something simple e.g. `<div><div` isn't a closed tag and flagged by the editor (similar to how Webstorm will do it). Thanks for the explanation. hopefully more requests for this will roll in but at this point I can't see it happening!

**Comment by jbenguira** [comment]
+1 this is really needed

**Comment by alexkoffler** [comment]
+1 jumping on the this is needed train

**Comment by JasonFritsche** [comment]
+1 Another vote for needed

**Comment by jainharsh97** [comment]
+1 Another vote for needed

**Comment by silli-scale** [comment]
+1 for this
